src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs

```diff
@@ -44,9 +43,9 @@ internal class ClusterDnsEndPointDiscoverer : IEndPointDiscoverer {
 			_maxDiscoverAttempts = maxDiscoverAttempts;
 			_gossipSeeds = gossipSeeds;
 			_gossipTimeout = gossipTimeout;
-			_client = new HttpAsyncClient(_gossipTimeout, httpMessageHandler, enableVersion5Compability);
+			_compatibilityMode = compatibilityMode;
+			_client = new HttpAsyncClient(_gossipTimeout, httpMessageHandler, _compatibilityMode);
```

Comment:
Why is `_compatibilityMode` passed in here? You can just condition the argument you pass to the usage of it (the hostHeader parameter is already there, defaulted to `""`).

Reply:
Because host header defaulted to an empty string is not enough. The user can still pass something else, hence why we are always overwriting the host header value if version 5 compatibility mode is enabled: https://github.com/EventStore/EventStore/blob/54726446a4ce5815953be1078318f8e58aff2598/src/EventStore.ClientAPI/Transport.Http/HttpAsyncClient.cs#L91

Reply:
It is enough.  There is no-where else that the compatibility mode is passed into `HttpAsyncClient` constructor, only in `ClusterDnsEndPointDiscoverer`.  By passing in empty string to the `Get` method, you achieve the exact same behaviour without needlessly bleeding the compatibility mode concept to `HttpAsyncClient`

Reply:
Also I feel like that host header compatibility flag should be better described as "MonoCompatibility" or similar since it doesn't affect non-mono servers?

Reply:
I would agree if `HttpAsyncClient` was a user-facing class. I would also argue passing the compatibility mode in this fashion conveys better why we actually have an empty string. You also keep overlooking the fact that having `hostHeader` defaulted to empty string doesn't prevent the user from overwriting that value from the connection setting https://github.com/EventStore/EventStore/blob/711ff91a6c7c6ef80afd488c5611f49b7d5ab232/src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs#L188

Reply:
Re. your last point, I didn't see that - What path can the user overwrite the header value?

Reply:
No worries, it happens when the user asks to use the DNS to resolve the gossip seeds: https://github.com/EventStore/EventStore/blob/711ff91a6c7c6ef80afd488c5611f49b7d5ab232/src/EventStore.ClientAPI/Internal/ClusterDnsEndPointDiscoverer.cs#L41

Reply:
About the naming, whether it's `MonoCompatibility` or a version number, I agree for the initial issue, your name is more precise. However, we introduce a more stable naming strategy that gives us room for further expansion by using the number version.

Reply:
Having said that, the auto compatibility mode you introduced in your PR is more future-proof and renders the naming strategy useless. So having `MonoCompatibility` might be a better solution after all :smile: !